core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreferWritePartitioning.java

```diff
@@ -46,6 +48,18 @@ public Result apply(TableWriterNode node, Captures captures, Context context)
             return Result.empty();
         }
 
+        double numberOfPartitions = node.getPreferredPartitioningScheme().get().getPartitioning().getColumns().stream()
+                .mapToDouble(column ->
```

Comment:
Please extract and reuse code from `io.trino.cost.AggregationStatsRule`. When we change computations for aggregation stats, we want to cover this place too.